storage/src/test/java/org/apache/kafka/tiered/storage/actions/ExpectEmptyRemoteStorageAction.java

```diff
@@ -36,8 +36,8 @@ public ExpectEmptyRemoteStorageAction(TopicPartition topicPartition) {
     public void doExecute(TieredStorageTestContext context) throws InterruptedException {
         TestUtils.waitForCondition(() -> {
             LocalTieredStorageSnapshot snapshot = context.takeTieredStorageSnapshot();
-            return !snapshot.getTopicPartitions().contains(topicPartition) &&
-                    snapshot.getFilesets(topicPartition).isEmpty();
+            // With KAFKA-15166, snapshot should not contain the topicPartition
```

Comment:
We don't differentiate the case between segment deletion and topic deletion so the underlying remote-storage-manager doesn't know when to remove any marker files/folders. In case of LocalTieredStorage, there will be empty partition directories.

Reply:
OK, so I think we should add all the explanation into the comment. From the current comment, it's not clear to me.